### PR TITLE
pr: Add testing

### PR DESCRIPTION
**Pull request by iainlane**
This is part of a _PR stack_:

  * #57
  * #58  + #62 <- you are here
  * #61

Some small refactoring to allow for it, nothing major.

### CODE REVIEW

File: cmd/wait-for-github/pr_test.go [hunk 1]
```diff
@@ -0,0 +1,262 @@
...
+		MergedAt: 1234567890,
+	}, gotCommitInfo)
+	require.Equal(t, fs.FileMode(0644), ffw.perm)
+}
+
+type erroringFileWriter struct{}
+
+func (e erroringFileWriter) WriteFile(filename string, data []byte, perm fs.FileMode) error {
+	return fmt.Errorf("error")
+}
```
Comment (zzehring) [inline cmd/wait-for-github/pr_test.go:172]: Super nit, but I think you could accomplish the same by just defining `erroringFileWriter` as a `fakeFileWriter` with an error. Reduces # lines of code slightly. Something like:

```suggestion
var erroringFileWriter := &fakeFileWriter{err:  fmt.Errorf("error")}
```

**Review by zzehring — APPROVED** [review]
LGTM! very small refactor nit, but definitely not needed.

File: cmd/wait-for-github/pr_test.go (hunk 1 above)
Comment (iainlane) [reply cmd/wait-for-github/pr_test.go:172]: Super nice, that works, I will change it, thanks!

Comment (iainlane) [reply cmd/wait-for-github/pr_test.go]: https://github.com/grafana/wait-for-github/compare/51e6764f27800fce2c9c7439f2fab6b7ad257147..356feac05a700ff6275254f2c1a9fe57f50867d6

going to merge now :muscle:

**Pull request merged**